samtranslator/swagger/swagger.py

```diff
@@ -1134,7 +1134,10 @@ def add_request_parameters_to_method(self, path, method_name, request_parameters
 
                 parameter_name = request_parameter["Name"]
                 location_name = parameter_name.replace("method.request.", "")
-                location, name = location_name.split(".")
+
+                parameter_name_parts = location_name.split(".")
+                location = parameter_name_parts[0]
+                name = ".".join(parameter_name_parts[1:])
```

Comment:
`split()` has a 2nd argument `maxsplit` which can be used here: https://docs.python.org/3/library/stdtypes.html#str.split